pkg/kvstore/kvstore.go

```diff
@@ -35,88 +36,128 @@ const (
 	BaseKeyPrefix = "cilium"
 )
 
-// Get returns value of key
-func Get(key string) ([]byte, error) {
-	v, err := Client().Get(key)
+// ExtendedKVStore is a struct that export multiple kvstore methods
+type ExtendedKVStore struct {
+	namespace string
+	mutex     lock.Mutex
+}
+
+// GetKVStoreExtendedClient returns a ExtendedKVStore struct
+func GetKVStoreExtendedClient() *ExtendedKVStore {
+	return &ExtendedKVStore{}
+}
+
+// GetKVStoreExtendedClientWithNamespace returns a ExtendedKVStore struct with the given namespace.
+func GetKVStoreExtendedClientWithNamespace(namespace string) *ExtendedKVStore {
```

Comment:
As this allocated a new struct, I'd call this `NewKVStoreExtendedClientWithNamespace()`.

Reply:
Now the `namespace` here only splits metrics to multiple sets, or namespaces. On the outset, looking at the modified KV-store API, one might think that the namespace somehow has an effect on the KV-store transactions (like putting the key-value pairs into distinct namespaces.

I don't know if there is a good solution to this, as calling it `MetricsNamespace` gets too long?

Reply:
2 cents, I see this kind of usage in the current PR:
```
kvClient = kvstore.GetKVStoreExtendedClientWithNamespace(subsys)
```

I think it would be just as expressive with more clarity to do something like:
```
kvClient = kvstore.NewClient().WithMetrics(subsys)
```